Mathlib/CategoryTheory/SmallObject/Iteration.lean

```diff
@@ -78,29 +79,26 @@ def coconeOfLE : Cocone (restrictionLT F hi) where
 
 end Iteration
 
-variable [IsWellOrder J (· < ·)] [OrderBot J]
-
 /-- The category of `j`th iterations of a functor `Φ` equipped with a natural
 transformation `ε : 𝟭 C ⟶ Φ`. An object consists of the data of all iterations
 of `Φ` for `i : J` such that `i ≤ j` (this is the field `F`). Such objects are
 equipped with data and properties which characterizes the iterations up to a unique
 isomorphism for the three types of elements: `⊥`, successors, limit elements. -/
-structure Iteration (j : J) where
+structure Iteration [Preorder J] [OrderBot J] [SuccOrder J] (j : J) where
   /-- The data of all `i`th iterations for `i : J` such that `i ≤ j`. -/
   F : { i // i ≤ j } ⥤ C ⥤ C
   /-- The zeroth iteration is the identity functor. -/
   isoZero : F.obj ⟨⊥, bot_le⟩ ≅ 𝟭 C
```

Comment:
I'm wondering whether we should replace ⊥ by IsMin ...

Reply:
This would fit more closely with the design of `IsSuccLimit`, but again, I'd rather leave that to a follow-up PR.